src/long.ts

```diff
@@ -1021,9 +1025,35 @@ export class Long {
     if (options && options.relaxed) return this.toNumber();
     return { $numberLong: this.toString() };
   }
-  static fromExtendedJSON(doc: { $numberLong: string }, options?: EJSONOptions): number | Long {
-    const result = Long.fromString(doc.$numberLong);
-    return options && options.relaxed ? result.toNumber() : result;
+  static fromExtendedJSON(
+    doc: { $numberLong: string },
+    options?: EJSONOptions
+  ): number | Long | bigint {
+    const { useBigInt64 = false, relaxed = true } = { ...options };
+
+    if (doc.$numberLong.length > MAX_INT64_STRING_LENGTH) {
+      throw new BSONError('int64 string is too long');
+    }
+
+    if (!DECIMAL_REG_EX.test(doc.$numberLong)) {
+      throw new BSONError('int64 string is not a valid decimal integer');
```

Comment:
```suggestion
      throw new BSONError(`int64 string "${doc.$numberLong}" is not a valid decimal integer`);
```

Still think we should put the input in the error message, makes it searchable when someone needs to go looking for the bad input

Reply:
Also going to do this in the `useBigInt64=true` case when the value provided is outside of int64 range.